build/Band_Representations.jl

```diff
@@ -0,0 +1,247 @@
+        end
+    end
+    return χ_G_k_G
+end
+
+function subduce_onto_littlegroups(k::String, kvec::KVec, ρ::Int64, sitegroup::SiteGroup{D}) where D
+    sgnum=sitegroup.num
+    num_lgirreps= size(get_lgirreps(sgnum, D)[k])[1]
+    
+    size_lgirreps=size(get_lgirreps(sgnum, D)[k][1].matrices)[1]
+
+    character_table=Matrix{Complex{Float64}}(undef, size_lgirreps, num_lgirreps )
+
+    for j in 1:num_lgirreps
+        for i in 1:size_lgirreps
+            character_table[i, j]=tr(get_lgirreps(sgnum, D)[k][j].matrices[i])
+        end
+    end
+
+    XVector=Matrix{Complex{Float64}}(undef, (size_lgirreps, 1))
+
+    for i in 1:size_lgirreps
+        rep_index=group_representative(group(get_lgirreps(sgnum, D)[k][1])[i], spacegroup(sgnum, D)) 
+        iter=induced_band_representation(sitegroup, spacegroup(sgnum, D)[rep_index], ρ, kvec)
+        XVector[i, 1]= iter
+    end
+
+    return round.(character_table\XVector)
+
+end
+
+function return_irvec(sgnum::Int64, ρ::Int64, sitegroup::SiteGroup{D}, maximal::Bool) where D
+    maximal ? klab2kv = Dict(klab => kvec(lg) for (klab, lg) in get_maximal_lgs(sgnum, D)) : klab2kv = Dict(klab => kvec(lg) for (klab, lg) in get_littlegroups(sgnum, D))
+    irvec = Int[]
+    for (klab, kv) in klab2kv
+        kv_array   = subduce_onto_littlegroups(klab, kv, ρ, sitegroup)
+        append!(irvec, Int.(kv_array))
+    end
+    return irvec
+end
+
+function return_irvec2d(sgnum::Integer, D::Integer=2, maximal::Bool=true)
+    sg=spacegroup(sgnum, D)
+    wps=get_wycks(sgnum, D)
+    A=Vector{Vector{Int}}(undef, 0)
+
+    sitegroups=SiteGroup.(Ref(sg), wps)
+
+    maximal_wyckoffs=Array{WyckPos{2}, 1}(undef, 0)
+
+    for maximal_sitegroup in findmaximal(sitegroups)
+        push!(maximal_wyckoffs, maximal_sitegroup.wp)
+    end
+
+    for wp in maximal_wyckoffs
+
+        sitegroup=SiteGroup(sg, wp)
+        ρ_max=size(site_symmetry_irreps(sitegroup))[1]
+        for ρ in 1:ρ_max
+            push!(A, return_irvec(sgnum, ρ, sitegroup, maximal))
+        end
+    end
+    return hcat(A...)
+end
+
+
+function get_maximal_lgirreps(sgnum::Integer, D::Integer)
+    maximal_lgirreps=Dict{String, Array{LGIrrep{D}, 1}}()
+    for (klab, lgirrep) in get_lgirreps(sgnum, D)
+        isspecial(kvec(lgirrep[1].lg)) || continue
+            push!(maximal_lgirreps, klab => lgirrep)
+    end
+    return maximal_lgirreps
+end
+
+function get_maximal_lgs(sgnum::Integer, D::Integer)
+    maximal_lgs=Dict{String, LittleGroup{D}}()
+    for (klab, lg) in get_littlegroups(sgnum, D)
+        isspecial(kvec(lg)) || continue
+            push!(maximal_lgs, klab => lg)
+    end
+    return maximal_lgs
+end
+
+function make_bandrep_set(sgnum::Integer; D::Integer=2, maximal=true)
+
+    maximal ? klab2kv = Dict(klab => kvec(lg) for (klab, lg) in get_maximal_lgs(sgnum, D)) : klab2kv = Dict(klab => kvec(lg) for (klab, lg) in get_littlegroups(sgnum, D))
+
+    irvec2d=return_irvec2d(sgnum, 2, maximal)
```

Comment:
Looks like we missed a `2` => `D`:
```suggestion
    irvec2d=return_irvec2d(sgnum, D, maximal)
```